### PR TITLE
add support for node v16

### PR DESCRIPTION
**Pull request by navarroaxel**
## Description
Please explain the changes you made here.

## Checklist

Please review this checklist before submitting a pull request.

- [ ] Code compiles correctly
- [ ] Created tests, if possible
- [x] All tests passing (`npm run test:all`)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by vladimyr** [comment]
I'm not eager on dropping v10 support either but I guess it can be safely done since it reached the EOL stage just recently: https://github.com/nodejs/Release/tree/14e3d74#end-of-life-releases

**Review by vladimyr — APPROVED** [review]

**Comment by agnivade** [comment]
Should we also update the min engine version here: https://github.com/tldr-pages/tldr-node-client/blob/3511d91454e1056d4c0926216a15044295734f0c/package.json#L25 to be above 10?

**Comment by vladimyr** [comment]
It should reflect tested versions, so `>=12` seems more appropriate.

**Comment by bl-ue** [comment]
Ha! It went into Maintenance LTS Start on 2020-05-19, one year ago today! 😮

**Review by owenvoke — APPROVED** [review]

**Review by bl-ue — APPROVED** [review]

**Comment by agnivade** [comment]
@vladimyr - Do you feel this needs a major version from the tool's perspective? Or it's okay to bump the minor version?

**Comment by vladimyr** [comment]
@agnivade Take a look at this example https://github.com/sindresorhus/open-cli/releases Bump from v5 to v6 was exactly due to a change in minimum supported node version so my answer is: yes we need a major version bump.

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
   ],
   "homepage": "http://tldr-pages.github.io",
   "engines": {
-    "node": ">=6.12.0"
+    "node": ">=12.0.0"
```
Comment (vladimyr) [inline package.json:25]:
```suggestion
    "node": ">=12"
```
You don't need to fully specify a version.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        node-version: [10.x, 12.x, 13.x, 14.x]
+        node-version: [12, 14, 16]
```
Comment (vladimyr) [inline .github/workflows/test.yml:16]: What about v13 and v15? They are not LTS versions but we still officially support them so they should really be a part of our test matrix?

**Review by vladimyr — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (bl-ue) [reply .github/workflows/test.yml:16]: But isn't v13 EOL?

Comment (vladimyr) [reply .github/workflows/test.yml:16]: It is but I don't think there is good reason to further complicate the version support scheme unless it is really necessary. Simple min required version should be enough.

Comment (navarroaxel) [reply .github/workflows/test.yml:16]: We can assume that just a little segment of enthusiastic use non-LTS versions, also the probabilities to fail in v13 or v15 is too small. And we don't have the resources to maintain that kind of builds, and it's ok big project doesn't officially support non-LTS versions of Ubuntu (could work or not).

**Pull request closed without merging**